x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/format_selector.tsx

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiFormRow, EuiFieldNumber, EuiComboBox } from '@elastic/eui';
+import { IndexPattern } from '../types';
+import { IndexPatternColumn } from '../indexpattern';
+import { IndexPatternDimensionPanelProps } from './dimension_panel';
+
+const supportedFormats: Record<string, { title: string }> = {
+  number: {
+    title: i18n.translate('xpack.lens.indexPattern.numberFormatLabel', {
+      defaultMessage: 'Number',
+    }),
+  },
+  percent: {
+    title: i18n.translate('xpack.lens.indexPattern.percentFormatLabel', {
+      defaultMessage: 'Percent',
+    }),
+  },
+  bytes: {
+    title: i18n.translate('xpack.lens.indexPattern.bytesFormatLabel', {
+      defaultMessage: 'Bytes (1024)',
+    }),
+  },
+};
+
+type FormatSelectorProps = Pick<IndexPatternDimensionPanelProps, 'data'> & {
+  selectedColumn: IndexPatternColumn;
+  currentIndexPattern: IndexPattern;
+  onChange: (newFormat?: { id: string; params?: Record<string, unknown> }) => void;
+};
+
+export function FormatSelector(props: FormatSelectorProps) {
+  const { selectedColumn, onChange } = props;
+
+  const currentFormat =
+    selectedColumn.params && 'format' in selectedColumn.params
+      ? selectedColumn.params.format
+      : undefined;
+
+  const selectedFormat = currentFormat?.id ? supportedFormats[currentFormat.id] : undefined;
+
+  const defaultOption = {
+    value: '',
+    label: i18n.translate('xpack.lens.indexPattern.defaultFormatLabel', {
+      defaultMessage: 'Default',
+    }),
+  };
+
+  const currentDecimals = currentFormat?.params?.decimals;
+
+  return (
+    <>
+      <EuiFormRow
+        label={i18n.translate('xpack.lens.indexPattern.columnFormatLabel', {
+          defaultMessage: 'Display format',
+        })}
+        display="rowCompressed"
+      >
+        <EuiComboBox
+          fullWidth
+          compressed
+          isClearable={false}
+          data-test-subj="indexPattern-dimension-format"
+          singleSelection={{ asPlainText: true }}
+          options={[defaultOption].concat(
+            Object.entries(supportedFormats).map(([id, format]) => ({
+              value: id || '',
```

Comment:
```suggestion
              value: id,
```

Don't need the `|| ''` here, since it could only potentially be triggered for the empty string anyway, since `id` is always a string.